src/plugins/ui_feature/data_cleanup.ts

```diff
@@ -206,4 +210,41 @@ export class DataCleanupPlugin extends UIPlugin {
     }
     return CommandResult.Success;
   }
+
+  private trimWhitespace() {
+    const zones = this.getters.getSelectedZones();
+    const sheetId = this.getters.getActiveSheetId();
+    let count = 0;
+    for (const zone of zones) {
+      positions(zone).map(({ col, row }) => {
+        count += this.trimCell({ col, row, sheetId }) ? 1 : 0;
```

Comment:
I'd add a line of docstring to `trimCell` to tell when it returns true/flase, otherwise it isn't clear what this line does.

TBH I would even delete the return, and add here a check "if(cell containse whitespace)".

Reply:
If I do what you tell me here, 
then I will write all the content of the trimCell function in the trimWhitespace function, 
this does not improve the reading nn more

Ok to add a docstring

Reply:
I agree with Adrien. `trimCell` could return the trimmed content (or the same content if there's nothing to trim).
And you would dispatch here.
That way, the business logic (of trimming) is also side-effect free, which is always a good thing
```ts
const trimmedContent = this.trim(cell.content)
if (trimmedContent !== cell.content) {
    count += 1;
    this.dispatch("UPDATE_CELL", {
      sheetId,
      col,
      row,
      content: trimmedContent,
    });
}
```